appcues/src/main/java/com/appcues/ui/extensions/ModifierExt.kt

```diff
@@ -66,9 +70,12 @@ internal fun Modifier.outerPrimitiveStyle(
  * eg. ButtonPrimitive, ImagePrimitive
  */
 internal fun Modifier.innerPrimitiveStyle(component: ExperiencePrimitive) = this.then(
-    with(component) {
-        Modifier.padding(style.getPaddings())
-    }
+    when (component) {
+        // image primitive needs to be clipped to bound to avoid
+        // content drawing overflow (side-by-side primitives)
+        is ImagePrimitive -> Modifier.clipToBounds()
```

Comment:
does this affect the use case of a negative margin at all? where content could lay over the top of the item above it, for example? If not, any harm in applying `clipToBounds()` to all primitive types vs. the Image special case here?

Reply:
Ithought of leaving it to image since we can save this extra clipping to any other unless we see its necessary. (the border stuff is coming up and might be good to revisit this if we need this applying for other types too